test/unit/app/services/asset_manager/attachment_updater_test.rb

```diff
@@ -4,281 +4,234 @@ class AssetManager::AttachmentUpdaterTest < ActiveSupport::TestCase
         end
-      end
-    end
-
-    describe "draft status updates" do
-      let(:draft) { true }
-      let(:unpublished) { false }
-      let(:replaced) { false }
 
-      before do
-        attachment_data.stubs(:draft?).returns(draft)
-        attachment_data.stubs(:unpublished?).returns(unpublished)
-        attachment_data.stubs(:replaced?).returns(replaced)
+        AssetManager::AttachmentUpdater.call(attachment.attachment_data, link_header: true)
       end
 
-      it "marks corresponding assets as draft" do
-        update_service.expects(:call)
-                        .with(original_asset.asset_manager_id, attachment_data, nil, { "draft" => true })
-        update_service.expects(:call)
-                      .with(thumbnail_asset.asset_manager_id, attachment_data, nil, { "draft" => true })
+      it "updates all assets to have an empty access_limited array when the draft is not access limited" do
+        expected_attribute_hash = {
+          "access_limited_organisation_ids" => [],
+        }
+
+        assets.each do |asset|
+          AssetManager::AssetUpdater.expects(:call).with(asset.asset_manager_id, attachment.attachment_data, nil, expected_attribute_hash)
+        end
 
-        updater.call(attachment_data, draft_status: true)
+        AssetManager::AttachmentUpdater.call(attachment.attachment_data, access_limited: true)
       end
 
-      context "and attachment is not draft" do
-        let(:draft) { false }
+      it "marks corresponding assets as not draft when the attachment has been replaced" do
+        attachment.attachment_data.replace_with!(AttachmentData.create!(file: File.open(fixture_path.join("sample.docx"))))
 
-        it "marks corresponding assets as not draft" do
-          update_service.expects(:call)
-                          .with(original_asset.asset_manager_id, attachment_data, nil, { "draft" => false })
-          update_service.expects(:call)
-                        .with(thumbnail_asset.asset_manager_id, attachment_data, nil, { "draft" => false })
+        expected_attribute_hash = {
+          "draft" => false,
+        }
 
-          updater.call(attachment_data, draft_status: true)
+        assets.each do |asset|
+          AssetManager::AssetUpdater.expects(:call).with(asset.asset_manager_id, attachment.attachment_data, nil, expected_attribute_hash)
         end
+
+        AssetManager::AttachmentUpdater.call(attachment.attachment_data, draft_status: true)
       end
+    end
 
-      context "and attachment is replaced" do
-        let(:replaced) { true }
+    context "when the attachment's attachable is a draft and is access limited" do
+      it "updates the access limited state of all assets" do
+        edition = create(:draft_news_article, :access_limited)
+        attachment = create(:file_attachment, attachable: edition)
+        assets = [
```

Comment:
This whole thing can be removed and you can loop straight through attachment.attachment_data.assets. Factory should have the assets.

Reply:
Thanks, I will remove these